system-int-test/test/cmr/system_int_test/ingest/variable_ingest_test.clj

```diff
@@ -245,3 +245,47 @@
               {:keys [status concept-id revision-id]} response]
           (is (= 200 status))
           (is (= 3 revision-id)))))))
+
+(deftest variable-update-test
+  (let [{token :token} (variable-util/setup-update-acl
+                        (s/context) "PROV1" "user1" "update-group")
+        concept (variable-util/make-variable-concept
+                 {:Name "var1"} {:native-id "var-to-be-updated"})
+        ;; ingest the variable with name var1
+        {var-concept-id :concept-id
+         initial-revision-id :revision-id} (variable-util/ingest-variable
+                                            concept
+                                            (variable-util/token-opts token))]
+    ;; sanity check
+    (is (mdb/concept-exists-in-mdb? var-concept-id initial-revision-id))
+    (is (= 1 initial-revision-id))
+
+    (testing "ingest of a variable with the same name but a different native id is OK"
+      (let [concept (variable-util/make-variable-concept
+                     {:Name "var1"} {:native-id "a-different-native-id"})
+            {:keys [concept-id revision-id]} (variable-util/ingest-variable
+                                              concept
+                                              (variable-util/token-opts token))]
+        (is (mdb/concept-exists-in-mdb? concept-id revision-id))
+        (is (= 1 revision-id))))
+
+    (let [;; variable with a different variable name, but the same native-id
+           concept (variable-util/make-variable-concept
+                    {:Name "new-var-name"} {:native-id "var-to-be-updated"})]
+      (testing "update the variable with a different variable name is not allowed"
```

Comment:
Note that this is exactly the reason we added native-ids for collections. Providers wanted to be able to update the identifier (dataset ID), but use the same concept so they didn't need to recreate all the granules.

So in this case if someone later decides that a variable has a bad name they'll have to delete the variable, create a new one with the better name, and then redo all of the associations. I assume they are ok with this.

Reply:
That's what I was told.